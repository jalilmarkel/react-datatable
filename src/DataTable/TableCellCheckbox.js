import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { useTableContext } from './DataTableContext';
import { CellBase } from './Cell';
import Checkbox from './Checkbox';

const TableCellCheckboxStyle = styled(CellBase)`
  flex: 0 0 48px;
  justify-content: center;
  align-items: center;
  user-select: none;
  white-space: nowrap;
`;

const TableCellCheckbox = ({ name, row, selected }) => {
  const {
    dispatch,
    data,
    keyField,
    selectableRowsComponent,
    selectableRowsComponentProps,
    selectableRowDisabled,
  } = useTableContext();
  const disabled = selectableRowDisabled && selectableRowDisabled(row);
  const rowCount = data.length;

  const handleOnRowSelected = useCallback(() => dispatch({
    type: 'SELECT_SINGLE_ROW',
    row,
    isSelected: selected,
    keyField,
    rowCount,
  }), [dispatch, row, selected, keyField, rowCount]);

  return (
    <TableCellCheckboxStyle
      onClick={e => e.stopPropagation()}
      className="rdt_TableCell"
      noPadding
    >
      <Checkbox
        name={name}
        component={selectableRowsComponent}
        componentOptions={selectableRowsComponentProps}
        checked={selected}
        aria-checked={selected}
        onClick={handleOnRowSelected}
        disabled={disabled}
      />
    </TableCellCheckboxStyle>
  );
};

TableCellCheckbox.propTypes = {
  name: PropTypes.string.isRequired,
  row: PropTypes.object.isRequired,
  selected: PropTypes.bool.isRequired,
};

export default TableCellCheckbox;
