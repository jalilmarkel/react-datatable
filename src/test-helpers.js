import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import theme from './themes/default';

export const renderWithTheme = (tree, ...args) =>
  render(<ThemeProvider theme={theme()}>{tree}</ThemeProvider>, ...args);

