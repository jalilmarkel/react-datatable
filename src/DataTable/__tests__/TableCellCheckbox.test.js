import 'jest-styled-components';
import React from 'react';
import { cleanup } from 'react-testing-library';
import { renderWithTheme } from '../../test-helpers';
import TableCellCheckbox from '../TableCellCheckbox';

afterEach(cleanup);

test('component <TableCellCheckbox /> should render correctly', () => {
  const { container } = renderWithTheme(<TableCellCheckbox row={{}} column={{ selector: 'test' }} onClick={() => { }} />);

  expect(container.firstChild).toMatchSnapshot();
});
