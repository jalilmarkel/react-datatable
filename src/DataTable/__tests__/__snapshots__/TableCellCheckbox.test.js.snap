// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 0 0 48px;
  -ms-flex: 0 0 48px;
  flex: 0 0 48px;
  font-size: 13px;
  color: rgba(0,0,0,.87);
  min-height: 48px;
}

<div
  class="rdt_TableCell c0"
>
  <input
    aria-label="select-row-456"
    name="select-row-456"
    style="font-size: 18px; cursor: pointer;"
    type="checkbox"
  />
</div>
`;

exports[`should render correctly if a selectableRowsComponent is provided 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 0 0 48px;
  -ms-flex: 0 0 48px;
  flex: 0 0 48px;
  font-size: 13px;
  color: rgba(0,0,0,.87);
  min-height: 48px;
}

<div
  class="rdt_TableCell c0"
>
  <div>
    hahah
  </div>
</div>
`;

exports[`should render correctly if a selectableRowsComponent is provided with selectableRowsComponentProps 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 0 0 48px;
  -ms-flex: 0 0 48px;
  flex: 0 0 48px;
  font-size: 13px;
  color: rgba(0,0,0,.87);
  min-height: 48px;
}

<div
  class="rdt_TableCell c0"
>
  <div>
    hahah
  </div>
</div>
`;
